### PR TITLE
Adds make_time ExprFunction

### PR DESCRIPTION
**Pull request by abhikuhikar**
*Issue #, if available:*

*Description of changes:*
 - Adds `make_time` `ExprFunction`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lang/src/org/partiql/lang/eval/builtins/MakeTimeExprFunction.kt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            it.bigDecimalValue()
+        }
+
+        val tzMinutes: Int? = when (args.size) {
+            4 -> args[3].let {
+                it.validateType(ExprValueType.INT)
+                it.intValue()
+            }
+            else -> null
+        }
```
Comment (almann) [inline lang/src/org/partiql/lang/eval/builtins/MakeTimeExprFunction.kt:61]: Is it expected that I have to have exact types?  Should it be?

**Review by almann — COMMENTED** [review]
Overall this seems good--though do we really need/want to have strict type requirements on numeric parameters?  I ask because I doubt the SQL implementation do, but we should verify this and be explicit about it.

File: lang/src/org/partiql/lang/eval/builtins/MakeTimeExprFunction.kt (hunk 1 above)
Comment (abhikuhikar) [reply lang/src/org/partiql/lang/eval/builtins/MakeTimeExprFunction.kt:61]: According to Postgresql, yes. Although we might be able to allow `int` as well as `decimal` for `second`.
However, the others strictly need to be `int` as per Postgresql. 
I think our `make_time` is inspired from Postgresql solely and hence we should follow the semantics closely as we can.

File: docs/user/BuiltInFunctions.md [hunk 2]
```diff
@@ -505,6 +505,39 @@ MAKE_DATE(missing, 02, 28)               -- null
 MAKE_DATE(2021, missing, 28)             -- null
 MAKE_DATE(2021, 02, missing)             -- null
 ```
+### MAKE_TIME
+
+Given the values for the hour (int), minute (int), second (BigDecimal) and optionally for timezone minutes (int) returns the associated time.
+
```
[the remark this answers is not in the document]
Comment (abhikuhikar) [reply docs/user/BuiltInFunctions.md:511]: Postgresql throws an error in case the arguments do not match the required types.

**Review by almann — APPROVED** [review]
:ship:

**Review by almann — APPROVED** [review]

**Pull request merged**